### ISSUE TITLE
TLS Support?

### ISSUE THREAD

**Issue by tonyespy** [edgexfoundry/device-rfid-llrp-go#32]
It's unclear from the top-level README.md file whether or not this device service supports TLS when communicating with a reader.

The latest [RFID specification](https://www.gs1.org/docs/epc/LLRP_standard_i2_r_2021-01-27.pdf) includes a section titled "Security in TCP Transport" which indicates that both server and client TLS authentication are possible...

**Note** - this issue is being reported as of the initial untagged version of the service, after it's recent promotion from holding.

**Comment by lenny-intel** [comment]
I pretty sure it doesn't. @ajcasagrande ??

**Comment by ajcasagrande** [comment]
You are correct that the spec details TLS communication on a default port of 5085. We do not currently support this functionality. As far as I knew Impinj readers did not support TLS, however doing some digging it appears like it does:


> **Octane 5.6.0**
> - Encrypt LLRP communications using transport layer security (TLS) protocol version 1.2
>  - Self-signed certificate (no client validation)
>  - Inbound connections only
>  - Cipher suites
>    - TLS_RSA_WITH_3DES_EDE_CBC-SHA
>    - TLS_RSA_WITH_AES_128_CBC-SHA
>    - TLS-RSA_WITH_AES_128_GCM-SHA256

I am not sure if we were to disable certificate checking and set the protocol and port properly if it would work or not.

**Comment by lenny-intel** [comment]
Question answered
